tests/cpu/test_seeding.cpp

```diff
@@ -41,27 +57,30 @@ TEST(seeding, case1) {
     // Adjust parameters
     finder_config.deltaRMax = 100.f * unit<float>::mm;
     finder_config.maxPtScattering = 0.5f * unit<float>::GeV;
-    traccc::seeding_algorithm sa(finder_config, grid_config, filter_config,
-                                 host_mr);
+    traccc::host::seeding_algorithm sa(finder_config, grid_config,
+                                       filter_config, host_mr);
 
-    spacepoint_collection_types::host spacepoints;
+    measurement_collection_types::host measurements(&host_mr);
+    edm::spacepoint_collection::host spacepoints{host_mr};
 
     // Spacepoints from 16.62 GeV muon
-    spacepoints.push_back({{36.6706f, 10.6472f, 104.131f}, {}});
-    spacepoints.push_back({{94.2191f, 29.6699f, 113.628f}, {}});
-    spacepoints.push_back({{149.805f, 47.9518f, 122.979f}, {}});
-    spacepoints.push_back({{218.514f, 70.3049f, 134.029f}, {}});
-    spacepoints.push_back({{275.359f, 88.668f, 143.378f}, {}});
+    add_spacepoint(measurements, spacepoints, {36.6706f, 10.6472f, 104.131f});
```

Comment:
If you want to push the idea of SoA for all EDMs, could you come up with a user-friendly interface in adding elements?  

For example, can we just keep the command the same as before?: 
`spacepoints.push_back({{36.6706f, 10.6472f, 104.131f}, {}});`

What I heard in the past is that something called a proxy enables this kind of interface.

Reply:
Indeed, this has been on my radar since a while. 🤔 Not sure yet exactly how to do it, but template magic should be able to reach that far... 🤔

Note though that in traccc this is mostly an inconvenience in the testing code. That's why it wasn't the highest priority for me so far. Because the core libraries don't need this functionality all that much.

But yes, I do want to make it nicer in the future. 👍

Reply:
The reason that this is mostly an inconvenience in the testing code is that our example codes are mostly running on the full chain. However, users might need to use only the part of algorithms depending on the experiments or their situations. IO or examples directory is not very recommended to be used by arbitrary experiments as well.

You can find tutorial codes to see what I expect with `traccc::core` application: https://github.com/beomki-yeo/traccc-tutorial/blob/main/tutorials/seeding.cpp